pkg/datapath/linux/config/config.go

```diff
@@ -472,7 +476,10 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 		cDefinesMap["ENABLE_JIFFIES"] = "1"
 	}
 
-	if option.Config.EnableIdentityMark {
+	if option.Config.EnableIdentityMark && !option.Config.EnableWireguard {
```

Comment:
Setting both to true seems to blindly disable the identity mark. Should we instead emit a warning or info message that identity marks are disabled?

Also it looks like the default for `EnableIdentityMark` is to be set `true`, so maybe just set it to `false` in case wireguard is enabled?

Reply:
I got rid of this check. Please see https://github.com/cilium/cilium/pull/15383#discussion_r602342047.